src/slipcover/slipcover.py

```diff
@@ -645,42 +668,51 @@ def register_module(self, m):
-                if self.replace_map:
-                    visited : set = set()
+            for file, new_set in newly_seen.items():
+                for co in self.instrumented[file]:
+                    self.deinstrument(co, new_set)
+
+                self.all_seen[file].update(new_set)
+
+            # Replace references to code
+            if self.replace_map:
+                visited = set()
+
+                # XXX the set of function objects could be pre-computed at register_module;
+                # also, the same could be done for functions objects in globals()
+                for m in self.modules:
+                    for f in Slipcover.find_functions(m.__dict__.values(), visited):
+                        if f.__code__ in self.replace_map:
+                            f.__code__ = self.replace_map[f.__code__]
+
+                globals_seen = []
+                for frame in sys._current_frames().values():
+                    while frame:
+                        if not frame.f_globals in globals_seen:
+                            globals_seen.append(frame.f_globals)
+                            for f in Slipcover.find_functions(frame.f_globals.values(), visited):
+                                if f.__code__ in self.replace_map:
+                                    f.__code__ = self.replace_map[f.__code__]
 
-                    # XXX the set of function objects could be pre-computed at register_module;
-                    # also, the same could be done for functions objects in globals()
-                    for m in self.modules:
-                        for f in Slipcover.find_functions(m.__dict__.values(), visited):
+                        for f in Slipcover.find_functions(frame.f_locals.values(), visited):
                             if f.__code__ in self.replace_map:
                                 f.__code__ = self.replace_map[f.__code__]
 
-                    globals_seen = []
-                    for frame in sys._current_frames().values():
-                        while frame:
-                            if not frame.f_globals in globals_seen:
-                                globals_seen.append(frame.f_globals)
-                                for f in Slipcover.find_functions(frame.f_globals.values(), visited):
-                                    if f.__code__ in self.replace_map:
-                                        f.__code__ = self.replace_map[f.__code__]
+                        frame = frame.f_back
+
+                # all references should have been replaced now... right?
+                self.replace_map.clear()
 
-                            for f in Slipcover.find_functions(frame.f_locals.values(), visited):
-                                if f.__code__ in self.replace_map:
-                                    f.__code__ = self.replace_map[f.__code__]
 
-                            frame = frame.f_back # type: ignore[assignment]
+@functools.lru_cache(None)
```

Comment:
without this cache, I saw a ~2s slowdown relative to the current impl.

5.9s without exclusions
6.1s with exclusions and cache
8.8s with exclusions no cache
105s with coverage